packages/host/app/components/matrix/login.gts

```diff
@@ -196,8 +196,11 @@ export default class Login extends Component<Signature> {
       throw e;
     }
     if (auth) {
-      await this.matrixService.start(auth);
-      await this.router.refresh();
+      await this.matrixService.start(auth, async () => {
+        // do this in a callback to the matrix service, otherwise our component
+        // is torn down immediately after this call
+        await this.router.refresh();
+      });
```

Comment:
Perhaps this could use a comment explaining what the effective timing is of this. i.e. what does "start" wait for? When does it call the callback? Does it wait for the callback to finish? What is the state of the system while these async ops are in progress?

Reply:
In https://github.com/cardstack/boxel/pull/1615/commits/7e927be1b4f2d54d415731685bf3a5b0db2f284c I moved the router refresh into the `MatrixService.start()` and provided a warning in the `login.gts` that statements after the `start()` will not be executed because we have yielded to `MatrixService.start` in the EC task.